src/Microsoft.ML.Transforms/TensorFlow/TensorflowUtils.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Transforms;
+using Microsoft.ML.Transforms.TensorFlow;
+
+namespace Microsoft.ML.Transforms.TensorFlow
+{
+    public class TensorflowUtils
+    {
+        internal static DataKind Tf2MlNetType(TFDataType type)
+        {
+            switch (type)
+            {
+                case TFDataType.Float:
+                    return DataKind.R4;
+                case TFDataType.Double:
+                    return DataKind.R8;
+                case TFDataType.Int32:
+                    return DataKind.I4;
+                case TFDataType.Int64:
+                    return DataKind.I8;
+                case TFDataType.UInt32:
+                    return DataKind.U4;
+                case TFDataType.UInt64:
+                    return DataKind.U8;
+                case TFDataType.Bool:
+                    return DataKind.Bool;
+                case TFDataType.String:
+                    return DataKind.TX;
+                default:
+                    throw new NotSupportedException("Tensorflow type not supported.");
+            }
+        }
+
+        internal static TFDataType MlNet2TfType(Type type)
+        {
+            if (type == typeof(float))
```

Comment:
>( [](start = 30, length = 1)

This could be code-gen'ed by the TT file.

Reply:
I was thinking that too. Let me see...:)

---
In reply to: [210747481](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210747481) [](ancestors = 210747481)

Reply:
Actually there is already a method  in TFSharp. I copied it.

---
In reply to: [210747970](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210747970) [](ancestors = 210747970,210747481)

Reply:
We should still plan on deleting all this TF# code and code-gen'ing it.  It's all boiler-plate copy/paste that can be driven off the type mapping.

---
In reply to: [210758078](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210758078) [](ancestors = 210758078,210747970,210747481)

Reply:
Agreed! 

---
In reply to: [210760965](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210760965) [](ancestors = 210760965,210758078,210747970,210747481)